cardano-db/src/Cardano/Db/Schema.hs

```diff
@@ -15,6 +15,8 @@
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE UndecidableInstances #-}
 
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```

Comment:
After fixing the `Pool` error above, in the same build situation as above, these errors fill up my terminal:
```
src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘blockProtoMinor’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘blockVrfKey’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘blockOpCert’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘slotLeaderPoolHashId’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘slotLeaderDescription’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘slotLeaderHash’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘slotLeaderDescription’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘slotLeaderHash’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘slotLeaderPoolHashId’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘poolHashView’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘poolHashHashRaw’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘schemaVersionStageTwo’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘schemaVersionStageThree’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘schemaVersionStageOne’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘schemaVersionStageThree’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘schemaVersionStageOne’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...

src/Cardano/Db/Schema.hs:48:1: error: [-Wname-shadowing, -Werror=name-shadowing]
    This binding for ‘schemaVersionStageTwo’ shadows the existing binding
      defined at src/Cardano/Db/Schema.hs:48:1
   |
48 | share
   | ^^^^^...
```

Reply:
Is your core public so I cam mess around with it?

Reply:
It is not, unfortunately. I may be able to give minimal reproduction if that would be useful!

Reply:
Yes, that would be very helpful.

Reply:
Here's a minimal project that has the same failures. The cabal.project file is largely copy-pasted from plutus-apps since the other project needs the plutus libs as well.

https://github.com/FinleyMcIlwaine/db-sync-mre

I get the failures with just a `cabal build`. Thanks for your help!

Reply:
Sorry, got distracted by other things.

I understand the fix for the `import Data.Pool (Pool)` fix, even though I am surprised if does not manifest in the `cardano-db-sync` repo.

However, I am completely befuddled by the name shadowing warning. It does not actually make sense to me. I am going to bash on this some more.

Reply:
Ok, by adding:
```
{-# OPTIONS_GHC -ddump-splices #-}
```
I can dump the expanded TH, but I really do not think its worth pursuing this.